src/Compilers/Core/Portable/Text/SourceTextComparer.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Text
+{
+    internal class SourceTextComparer : IEqualityComparer<SourceText>
+    {
+        public static SourceTextComparer Instance = new SourceTextComparer();
+
+        public bool Equals(SourceText x, SourceText y)
+        {
+            if (x == null)
+            {
+                return y == null;
+            }
+            else if (y == null)
+            {
+                return false;
+            }
+
+            return x.ContentEquals(y);
+        }
+
+        public int GetHashCode(SourceText obj)
+        {
+            var checksum = obj.GetChecksum(useDefaultEncodingIfNull: true);
```

Comment:
Maybe all of these computations are fast and so it doesn't matter, but if the checksum hash is nonzero, is that good enough?